dotnet test --filter FullyQualifiedName fails to work with generic test classes

`dotnet test --filter "FullyQualifiedName=MyNamespace.MyTestsClass<ParameterType1,ParameterType2>.MyTestMethod"
`
Error message:
```
MSBUILD : error MSB1006: Property is not valid.
Switch: ParameterType2>.MyTestMethod

For switch syntax, type "MSBuild -help"

```

There is no way to escape comma. So there is no way to pass full qualified name if the tests class is generic with multiple parameter types.
Either put into documentation the explanation how to accomplish that, or report a bug to .NET Core developers team.

I need it because I run the same UI Tests code under both AndroidDriver and IOSDriver which I provide as generics parameters.

edit by @mairaw: added topic details
 
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 61ca4064-7481-a263-60fc-c70127f2ad16
* Version Independent ID: 2878af44-8208-7bd8-46d4-3acda3e82ce0
* Content: [dotnet test command - .NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-test?tabs=netcore21)
* Content Source: [docs/core/tools/dotnet-test.md](https://github.com/dotnet/docs/blob/master/docs/core/tools/dotnet-test.md)
* Product: **dotnet-core**
* Technology: **dotnet-cli**
* GitHub Login: @mairaw
* Microsoft Alias: **mairaw**

There is a way to [escape](https://docs.microsoft.com/en-us/visualstudio/msbuild/how-to-escape-special-characters-in-msbuild?view=vs-2019) the comma in MSBuild command-line property definitions: replace it with `%2C`.

```sh-session
$ dotnet new mstest
...
$  dotnet test --filter "FullyQualifiedName=MyNamespace.MyTestsClass<ParameterType1,ParameterType2>.MyTestMethod"
MSBUILD : error MSB1006: Property is not valid.
Switch: ParameterType2>.MyTestMethod

For switch syntax, type "MSBuild -help"
$ dotnet test --filter "FullyQualifiedName=MyNamespace.MyTestsClass<ParameterType1%2CParameterType2>.MyTestMethod"
Test run for S:\repro\dotnet\docs\issues\13749\bin\Debug\netcoreapp3.0\13749.dll(.NETCoreApp,Version=v3.0)
Microsoft (R) Test Execution Command Line Tool Version 16.3.0-preview-20190808-04
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...

A total of 1 test files matched the specified pattern.
No test matches the given testcase filter `FullyQualifiedName=MyNamespace.MyTestsClass<ParameterType1,ParameterType2>.MyTestMethod` in S:\repro\dotnet\docs\issues\13749\bin\Debug\netcoreapp3.0\13749.dll
```

It would be nice if `dotnet test` did this automatically when handling the `--filter` argument. I filed dotnet/cli#12233 to track that.

--- a/docs/core/testing/selective-unit-tests.md
+++ b/docs/core/testing/selective-unit-tests.md
@@ -15,7 +15,7 @@ Using filters that include exclamation mark (!) on `*nix` requires escaping sinc
 skips all tests if the namespace contains IntegrationTests: `dotnet test --filter FullyQualifiedName\!~IntegrationTests`.
 Note the backslash that precedes the exclamation mark.
 
-For FullyQualifiedName values that include a comma, as for generic type parameters, escape the comma with `%2C`. For example:
+For `FullyQualifiedName` values that include a comma for generic type parameters, escape the comma with `%2C`. For example:
 
 ```dotnetcli
 dotnet test --filter "FullyQualifiedName=MyNamespace.MyTestsClass<ParameterType1%2CParameterType2>.MyTestMethod"

--- a/docs/core/testing/selective-unit-tests.md
+++ b/docs/core/testing/selective-unit-tests.md
@@ -2,17 +2,24 @@
 title: Running selective unit tests
 description: How to use a filter expression to run selective unit tests with the dotnet test command in .NET Core.
 author: smadala
-ms.date: 03/22/2017
+ms.date: 04/29/2020
 ---
 
 # Running selective unit tests
 
 With the `dotnet test` command in .NET Core, you can use a filter expression to run selective tests. This article demonstrates how to filter which test are run. The following examples use `dotnet test`. If you're using `vstest.console.exe`, replace `--filter` with `--testcasefilter:`.
 
-> [!NOTE]
-> Using filters that include exclamation mark (!) on `*nix` requires escaping since `!` is reserved. For example, this filter
-> skips all tests if the namespace contains IntegrationTests: `dotnet test --filter FullyQualifiedName\!~IntegrationTests`.
-> Note the backslash that precedes the exclamation mark.
+## Character escaping
+
+Using filters that include exclamation mark (!) on `*nix` requires escaping since `!` is reserved. For example, this filter
+skips all tests if the namespace contains IntegrationTests: `dotnet test --filter FullyQualifiedName\!~IntegrationTests`.
+Note the backslash that precedes the exclamation mark.
+
+For FullyQualifiedName values that include a comma, as for generic type parameters, escape the comma with `%2C`. For example:
+
+```dotnetcli
+dotnet test --filter "FullyQualifiedName=MyNamespace.MyTestsClass<ParameterType1%2CParameterType2>.MyTestMethod"
+```
 
 ## MSTest
 

--- a/docs/core/testing/selective-unit-tests.md
+++ b/docs/core/testing/selective-unit-tests.md
@@ -151,3 +151,5 @@ namespace NUnitNamespace
 | <code>dotnet test --filter "FullyQualifiedName~UnitTest1&#124;TestCategory=CategoryA"</code> | Runs tests which have `UnitTest1` in `FullyQualifiedName` **or** `TestCategory` is `CategoryA`. |
 | `dotnet test --filter "FullyQualifiedName~UnitTest1&TestCategory=CategoryA"` | Runs tests which have `UnitTest1` in `FullyQualifiedName` **and** `TestCategory` is `CategoryA`. |
 | <code>dotnet test --filter "(FullyQualifiedName~UnitTest1&TestCategory=CategoryA)&#124;Priority=1"</code> | Runs tests which have either `FullyQualifiedName` containing `UnitTest1` **and** `TestCategory` is `CategoryA` **or** `Priority` is 1. |
+
+For more information, see [TestCase filter](https://github.com/Microsoft/vstest-docs/blob/master/docs/filter.md).

--- a/docs/core/tools/dotnet-test.md
+++ b/docs/core/tools/dotnet-test.md
@@ -115,7 +115,10 @@ Test projects specify the test runner using an ordinary `<PackageReference>` ele
 
 - **`-s|--settings <SETTINGS_FILE>`**
 
-  The `.runsettings` file to use for running the tests. [Configure unit tests by using a `.runsettings` file.](/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file) Note that the `TargetPlatform` element (x86|x64) has no effect for `dotnet test`. To run tests that target x86, install the x86 version of .NET Core. The bitness of the *dotnet.exe* that is on the path is what will be used for running tests.
+  The `.runsettings` file to use for running the tests. Note that the `TargetPlatform` element (x86|x64) has no effect for `dotnet test`. To run tests that target x86, install the x86 version of .NET Core. The bitness of the *dotnet.exe* that is on the path is what will be used for running tests. for more information, see the following resources:
+
+  - [Configure unit tests by using a `.runsettings` file.](/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file)
+  - [Configure a test run](https://github.com/Microsoft/vstest-docs/blob/master/docs/configure.md)
 
 - **`-t|--list-tests`**
 

--- a/docs/core/tools/dotnet-test.md
+++ b/docs/core/tools/dotnet-test.md
@@ -31,7 +31,7 @@ dotnet test -h|--help
 
 ## Description
 
-The `dotnet test` command is used to execute unit tests in a given project. The `dotnet test` command launches the test runner console application specified for a project. The test runner executes the tests defined for a unit test framework (for example, MSTest, NUnit, or xUnit) and reports the success or failure of each test. If all tests are successful, the test runner returns 0 as an exit code; otherwise if any test fails, it returns 1. The test runner and the unit test library are packaged as NuGet packages and are restored as ordinary dependencies for the project.
+The `dotnet test` command is used to execute unit tests in a given project. The `dotnet test` command launches the test runner console application specified for a project. The test runner executes the tests defined for a unit test framework (for example, MSTest, NUnit, or xUnit) and reports the success or failure of each test. If all tests are successful, the test runner returns 0 as an exit code; otherwise if any test fails, it returns 1. For multi-targeted projects, tests are run for each targeted framework. The test runner and the unit test library are packaged as NuGet packages and are restored as ordinary dependencies for the project.
 
 Test projects specify the test runner using an ordinary `<PackageReference>` element, as seen in the following sample project file:
 

--- a/docs/core/tools/dotnet-test.md
+++ b/docs/core/tools/dotnet-test.md
@@ -1,7 +1,7 @@
 ---
 title: dotnet test command
 description: The dotnet test command is used to execute unit tests in a given project.
-ms.date: 02/27/2020
+ms.date: 04/29/2020
 ---
 # dotnet test
 
@@ -115,7 +115,7 @@ Test projects specify the test runner using an ordinary `<PackageReference>` ele
 
 - **`-s|--settings <SETTINGS_FILE>`**
 
-  The `.runsettings` file to use for running the tests. [Configure unit tests by using a `.runsettings` file.](/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file)
+  The `.runsettings` file to use for running the tests. [Configure unit tests by using a `.runsettings` file.](/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file) Note that the `TargetPlatform` element (x86|x64) has no effect for `dotnet test`. To run tests that target x86, install the x86 version of .NET Core. The bitness of the *dotnet.exe* that is on the path is what will be used for running tests.
 
 - **`-t|--list-tests`**
 

--- a/docs/core/tools/dotnet-test.md
+++ b/docs/core/tools/dotnet-test.md
@@ -103,7 +103,7 @@ Test projects specify the test runner using an ordinary `<PackageReference>` ele
 
 - **`-o|--output <OUTPUT_DIRECTORY>`**
 
-  Directory in which to find the binaries to run. If not specified, the default path is `./bin/<configuration>/<framework>/`.  For projects with multiple target frameworks (via the `TargetFrameworks` property), you also need to define `--framework` when you specify this option.
+  Directory in which to find the binaries to run. If not specified, the default path is `./bin/<configuration>/<framework>/`.  For projects with multiple target frameworks (via the `TargetFrameworks` property), you also need to define `--framework` when you specify this option. `dotnet test` always run tests from the output directory. You can use <xref:System.AppDomain.BaseDirectory%2A?displayProperty=nameWithType> to consume test assets in the output directory.
 
 - **`-r|--results-directory <PATH>`**
 

